### PR TITLE
[Event] Add method to request calendarView from Office365 API

### PR DESCRIPTION
**Pull request by krichprollsch**
Since the getlist do not return all recurring instances of an event,
calendarView gives a way to get the occurrences, exceptions, and single
instances of events

### CODE REVIEW

File: src/Api/EventApi.php [hunk 1]
```diff
@@ -73,21 +100,57 @@ public function getList(Calendar $calendar = null, $filter = '', $orderBy = '')
...
+     * @param DateTime $to The date and time when the event ends.
+     * @param string $filter `$filter` query parameter to give to the request
+     * @param string $orderBy `$orderBy` query, to have an order of elements
+     *
+     * @see https://msdn.microsoft.com/office/office365/APi/calendar-rest-operations#GetCalendarView
+     */
+    public function getCalendarView(
+        Calendar $calendar = null,
+        DateTime $from = null,
+        DateTime $to = null,
```
Comment (Taluu) [inline src/Api/EventApi.php]: these shouldn't be null, this is not the scope of this adapter to make things such as "magically pick an arbitrary date"

File: src/Api/EventApi.php [hunk 2]
```diff
@@ -56,7 +78,12 @@ public function getList(Calendar $calendar = null, $filter = '', $orderBy = '')
         $url = 'events';
 
         if (null !== $calendar) {
-            $url = sprintf('calendars/%s/events', $calendar->getId());
+            $url = sprintf(
+                'calendars/%s/calendarview?startDateTime=%s&endDateTime=%s',
+                $calendar->getId(),
+                (new \DateTime('now'))->sub(new \DateInterval('P1Y'))->format('Y-m-d\TH:i:s\Z'),
+                (new \DateTime('now'))->add(new \DateInterval('P1Y'))->format('Y-m-d\TH:i:s\Z')
+            );
```
Comment (Taluu) [inline src/Api/EventApi.php]: the `getList` shouldn't arbitrarly use the `calendarView` if there is a calendar

Comment (krichprollsch) [inline src/Api/EventApi.php]: My bad, test I forgot to remove :/

File: src/Api/EventApi.php (hunk 1 above)
Comment (krichprollsch) [inline src/Api/EventApi.php]: +1

Comment (krichprollsch) [inline src/Api/EventApi.php]: fixed

File: src/Api/EventApi.php (hunk 2 above)
Comment (krichprollsch) [inline src/Api/EventApi.php]: fixed

File: src/Api/EventApi.php [hunk 3]
```diff
@@ -73,21 +94,54 @@ public function getList(Calendar $calendar = null, $filter = '', $orderBy = '')
...
-        if (200 > $response->getStatusCode() || 300 <= $response->getStatusCode()) {
-            throw new ApiErrorException($response);
+    /**
+     * Get the occurrences, exceptions, and single instances of events in a calendar view
+     *
+     * If a calendar is given, it fetches the events for this calendar ;
+     * otherwise, it takes the primary
+     *
+     * If no from is given, it fetches one year before now
+     * If no to is given, it fetches one year after now
```
Comment (Taluu) [inline src/Api/EventApi.php]: these comments are outdated

**Comment by Taluu** [comment]
looks good. Please remove the comments about the date and it should be mergeable

File: src/Api/EventApi.php (hunk 3 above)
Comment (krichprollsch) [inline src/Api/EventApi.php]: done

**Pull request merged**